service/matching/taskListManager.go

```diff
@@ -534,6 +562,61 @@ func (c *taskListManagerImpl) isFowardingAllowed(taskList *taskListID, kind type
 	return !taskList.IsRoot() && kind != types.TaskListKindSticky
 }
 
+func (c *taskListManagerImpl) isIsolationMatcherEnabled() bool {
+	return c.taskListKind != types.TaskListKindSticky && c.enableIsolation
+}
+
+func (c *taskListManagerImpl) shouldReload() bool {
+	return c.config.EnableTasklistIsolation() != c.enableIsolation && c.taskListKind != types.TaskListKindSticky
+}
+
+func (c *taskListManagerImpl) getIsolationGroupForTask(ctx context.Context, taskInfo *persistence.TaskInfo) (string, error) {
+	if c.enableIsolation && len(taskInfo.PartitionConfig) > 0 {
+		partitionConfig := make(map[string]string)
```

Comment:
nit: length known ahead of time: `make(map[string]string, len(taskinfo.PartitionConfig))`